multicluster/service-mirror/cluster_watcher.go

```diff
@@ -192,18 +192,30 @@ func (rcsw *RemoteClusterServiceWatcher) originalResourceName(mirroredName strin
 	return strings.TrimSuffix(mirroredName, fmt.Sprintf("-%s", rcsw.link.TargetClusterName))
 }
 
-func (rcsw *RemoteClusterServiceWatcher) getMirroredServiceLabels() map[string]string {
-	return map[string]string{
-		consts.MirroredResourceLabel:  "true",
-		consts.RemoteClusterNameLabel: rcsw.link.TargetClusterName,
+func (rcsw *RemoteClusterServiceWatcher) getMirroredServiceLabels(remoteService *corev1.Service) map[string]string {
+	labels := make(map[string]string)
+
+	if remoteService != nil {
+		for key, value := range remoteService.ObjectMeta.Labels {
+			if strings.HasPrefix(key, consts.SvcMirrorPrefix) {
+				continue
+			}
+			labels[key] = value
+		}
 	}
+	labels[consts.MirroredResourceLabel] = "true"
+	labels[consts.RemoteClusterNameLabel] = rcsw.link.TargetClusterName
+
+	return labels
 }
 
 func (rcsw *RemoteClusterServiceWatcher) getMirroredServiceAnnotations(remoteService *corev1.Service) map[string]string {
-	annotations := map[string]string{
-		consts.RemoteResourceVersionAnnotation: remoteService.ResourceVersion, // needed to detect real changes
-		consts.RemoteServiceFqName:             fmt.Sprintf("%s.%s.svc.%s", remoteService.Name, remoteService.Namespace, rcsw.link.TargetClusterDomain),
+	annotations := make(map[string]string)
+	for key, value := range remoteService.ObjectMeta.Annotations {
+		annotations[key] = value
 	}
+	annotations[consts.RemoteResourceVersionAnnotation] = remoteService.ResourceVersion // needed to detect real changes
+	annotations[consts.RemoteServiceFqName] = fmt.Sprintf("%s.%s.svc.%s", remoteService.Name, remoteService.Namespace, rcsw.link.TargetClusterDomain)
 
 	value, ok := remoteService.GetAnnotations()[consts.ProxyOpaquePortsAnnotation]
```

Comment:
Is this still necessary? It seems we are copying everything on L213-L215, so this might be redundant now.

Reply:
Tried it, but gives empty annotation maps. We'll have to modify the `cluster_watcher_test_util.go` to remove the expected annotations that were set before, but if these are necessary, then we need these lines.